js/components/common/react_modal.js.jsx

```diff
@@ -32,20 +35,88 @@ class CourseModal extends React.Component {
     this.state = {
       courseId: "",
       course: [],
-      sessions: [],
+      sessions: {},
       courseTitle: "",
     }
   }
 
+  /**
+   * Generate the data needed for each row of the timetable based on course meeting times for
+   * each session F, S, Y.
+   */
+  getTable(allMeetingTimes, session) {
+    const sessions = allMeetingTimes.filter(lec => lec.meetData.session === session)
+    const sortedSessions = sessions.sort((firstLec, secondLec) =>
+      firstLec.meetData.section > secondLec.meetData.section ? 1 : -1
+    )
+
+    let allData = []
+    sortedSessions.forEach(lecture => {
+      const occurrences = { times: [], rooms: [] }
+      const sortedTimeData = lecture.timeData.sort((occ1, occ2) =>
+        occ1.weekDay > occ2.weekDay ? 1 : -1
+      )
+      sortedTimeData.map(occurrence => {
+        let firstRoom = ""
+        if (occurrence.firstRoom === null || occurrence.firstRoom === undefined) {
+          firstRoom = " "
+        } else {
+          firstRoom = occurrence.firstRoom.room
+        }
+
+        let secondRoom = ""
+        if (occurrence.secondRoom === null || occurrence.secondRoom === undefined) {
+          secondRoom = " "
+        } else {
+          secondRoom = occurrence.secondRoom.room
+        }
+
+        if ((firstRoom != " ") & (secondRoom != " ")) {
+          firstRoom += ", "
+        }
+        occurrences.rooms.push(firstRoom + secondRoom)
+        occurrences.times.push(
+          convertToDay[occurrence.weekDay] +
+            "  " +
+            occurrence.startHour +
+            " - " +
+            occurrence.endHour
+        )
+      })
+      const rowData = {
+        Activity: lecture.meetData.section,
```

Comment:
Begin these keys with lowercase letters rather than uppercase letters. You can customize the header names in the column data passed to the `AgGridReact` component below.